query-engine/driver-adapters/src/wasm/conversion.rs

```diff
@@ -0,0 +1,52 @@
+use crate::conversion::JSArg;
+
+use super::to_js::{serde_serialize, ToJsValue};
+use crate::types::Query;
+use js_sys::{Array, JsString, Object, Reflect, Uint8Array};
+use wasm_bindgen::{prelude::wasm_bindgen, JsValue};
+
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen(extends = Object)]
+    pub type Buffer;
+
+    #[wasm_bindgen(static_method_of = Buffer)]
+    pub fn from(array: &Uint8Array) -> Buffer;
+}
+
+impl ToJsValue for Query {
+    fn to_js_value(&self) -> Result<wasm_bindgen::prelude::JsValue, wasm_bindgen::prelude::JsValue> {
+        let object = Object::new();
+        let sql = self.sql.to_js_value()?;
+        Reflect::set(&object, &JsValue::from(JsString::from("sql")), &sql)?;
+        let args = Array::new();
+        for arg in &self.args {
+            let value = arg.to_js_value()?;
+            args.push(&value);
+        }
+        Reflect::set(&object, &JsValue::from(JsString::from("args")), &args)?;
+
+        Ok(JsValue::from(object))
+    }
+}
```

Comment:
I understand the need for this `js_sys`-based approach, I'd like to try this out locally before merging as is, if that's ok.

Reply:
Sure, go ahead